src/coordinator/ts/m3db/block/converter.go

```diff
@@ -31,45 +35,88 @@ var (
 )
 
 // SeriesBlockToMultiSeriesBlocks converts M3DB blocks to multi series blocks
-func SeriesBlockToMultiSeriesBlocks(multiNamespaceSeriesList []MultiNamespaceSeries, seriesIteratorsPool encoding.MutableSeriesIteratorsPool) (MultiSeriesBlocks, error) {
+func SeriesBlockToMultiSeriesBlocks(
+	multiNamespaceSeriesList []MultiNamespaceSeries, seriesIteratorsPool encoding.MutableSeriesIteratorsPool, stepSize time.Duration) (MultiSeriesBlocks, error) {
 	// todo(braskin): validate blocks size and aligment per namespace before creating []MultiNamespaceSeries
 	var multiSeriesBlocks MultiSeriesBlocks
-	for multiNamespaceSeriesIdx, multiNamespaceSeries := range multiNamespaceSeriesList {
-		consolidatedSeriesBlocks, err := newConsolidatedSeriesBlocks(multiNamespaceSeries, seriesIteratorsPool)
+
+	for seriesIdx, multiNamespaceSeries := range multiNamespaceSeriesList {
+		consolidatedSeriesBlocks, err := newConsolidatedSeriesBlocks(multiNamespaceSeries, seriesIteratorsPool, stepSize)
 		if err != nil {
 			return nil, err
 		}
 
 		// once we get the length of consolidatedSeriesBlocks, we can create a
 		// MultiSeriesBlocks list with the proper size
-		if multiNamespaceSeriesIdx == 0 {
+		if seriesIdx == 0 {
 			multiSeriesBlocks = make(MultiSeriesBlocks, len(consolidatedSeriesBlocks))
 		}
 
 		for consolidatedSeriesBlockIdx, consolidatedSeriesBlock := range consolidatedSeriesBlocks {
+
 			// we only want to set the start and end times once
-			if multiNamespaceSeriesIdx == 0 {
-				multiSeriesBlocks[consolidatedSeriesBlockIdx].Start = consolidatedSeriesBlock.Start
-				multiSeriesBlocks[consolidatedSeriesBlockIdx].End = consolidatedSeriesBlock.End
+			if seriesIdx == 0 {
+				blockBounds := multiSeriesBlocks[consolidatedSeriesBlockIdx].Metadata.Bounds
+				blockBounds.StepSize = consolidatedSeriesBlock.Metadata.Bounds.StepSize
+				blockBounds.Start = consolidatedSeriesBlock.Metadata.Bounds.Start
+				blockBounds.End = consolidatedSeriesBlock.Metadata.Bounds.End
+				multiSeriesBlocks[consolidatedSeriesBlockIdx].Metadata.Bounds = blockBounds
+			}
+
+			dupedTags := consolidatedSeriesBlock.ConsolidatedNSBlocks[0].SeriesIterators.Iters()[0].Tags().Duplicate()
```

Comment:
Can you add comments here, or add a method on consolidatedNSBlocks that gives tags? There's a lot of indexing here without any clear guarantees that it's safe to do